iOS_SDK/OneSignalSDK/UnitTests/UIApplicationDelegateSwizzlingTests.m

```diff
@@ -408,6 +439,36 @@ - (void)testDoubleSwizzleInfiniteLoop {
     [localOrignalDelegate applicationWillTerminate:UIApplication.sharedApplication];
 }
 
+- (void)testAppDelegateSubClassIsNotSwizzledTwice {
+    // 1. Create a new delegate and assign it
+    id myAppDelegate = [AppDelegateForInfiniteLoopWithAnotherSwizzlerTest new];
+    UIApplication.sharedApplication.delegate = myAppDelegate;
+    
+    // 2. Create another Library's app delegate and assign it then swizzle
+    id thierAppDelegate = [OtherLibraryASwizzler new];
+    UIApplication.sharedApplication.delegate = thierAppDelegate;
+    [OtherLibraryASwizzler swizzleAppDelegate];
+    
+    // 3. Create another Library's app delegate subclass and assign it then swizzle
+    id thierAppDelegateSubClass = [OtherLibraryBSwizzlerSubClass new];
+    UIApplication.sharedApplication.delegate = thierAppDelegateSubClass;
+    [OtherLibraryBSwizzlerSubClass swizzleAppDelegate];
+    
+    // 4. Call something to confirm we don't get stuck in an infinite call loop
+    id<UIApplicationDelegate> delegate = UIApplication.sharedApplication.delegate;
+    [delegate applicationWillTerminate:UIApplication.sharedApplication];
+    
+    // 5. Ensure OneSignal's selector is called.
+    XCTAssertEqual([OneSignalAppDelegateOverrider
+        callCountForSelector:@"oneSignalApplicationWillTerminate:"], 1);
+    
+    // 6. Ensure other library selector is still called too.
+    XCTAssertTrue([OtherLibraryASwizzler selectorCalled]);
+    
+    // 7. Ensure other library subclass selector is still called too.
+    XCTAssertTrue([OtherLibraryASwizzler selectorCalled]);
```

Comment:
This is the exact same assert has above, did you mean to test `OtherLibraryBSwizzler` here?

Reply:
Yes good catch!